includes/Services/FlowService.php

```diff
@@ -0,0 +1,191 @@
+						foreach($value as $index_key => $index_value) {
+
+							// For Indexed Arrays having values as an Array: Indexed/Associative
+							if(is_array($index_value)) {
+								if(empty($flow_data[$key][$index_key]))
+									$updated_flow_data[$key][$index_key] = $index_value;
+								else
+									$updated_flow_data[$key][$index_key] = $flow_data[$key][$index_key];
+							}
+
+							// For Indexed Arrays having values as a String/Boolean
+							else {
+								if(empty($flow_data[$key]))
+									$updated_flow_data[$key] = $value;
+								else
+									$updated_flow_data[$key] = $flow_data[$key];
+							}
+						}
+					}
+					else
+						$updated_flow_data[$key] = self::update_flow_data_recursive($value, $flow_data[$key], $updated_flow_data[$key]);
+				}
+			}
+			return $updated_flow_data;
+		  }
+	}
+
+	/*
+	 * function to update the Database recursively based on Values opted or entered by the User
+	 */	
+	public static function update_post_call_data_recursive(&$flow_data, &$params) {
+		{
+			foreach ($flow_data as $key => $value)
+			{ 
+				// Updates value entered by the user
+				if (isset($params) && array_key_exists($key, $params) && !is_array($value))
+					$flow_data[$key] = $params[$key];
+
+				// Retains the Blueprint Value if no input from the User
+				else $flow_data[$key] = $value;
+								
+				if ( is_array( $value ) ) {
+					// To handle Indexed Arrays gracefully
+					if (isset($params[$key]) && count(array_filter(array_keys($params[$key]), 'is_string')) === 0 && !empty($params[$key])) 
+						$flow_data[$key] = $params[$key];
+					else
+						$flow_data[$key] = self::update_post_call_data_recursive($value, $params[$key]);
+				}
+			}
+			return $flow_data;
+		}
+	}
+
+    /*
+	 * Read onboarding flow options from database
+	*/
+	public static function read_flow_data_from_wp_option() {
+		return \get_option( Options::get_option_name( 'flow' ), false );
+	}
+
+    /*
+	 * add onboarding flow options
+	 */
+	private static function save_details_to_wp_options( $data ) {
+		return \add_option( Options::get_option_name( 'flow' ), $data );
+	}
+
+	/*
+	 * update onboarding flow options
+	 */
+	public static function update_wp_options_data_in_database( $data ) {
+		return \update_option( Options::get_option_name( 'flow' ), $data );
+	}
+
+    /**
+	 * check the current flow type and update default data if flowtype is ecommerce.
+	 *
+	 * @param default flow data.
+	 *
+	 * @return array
+	 */
+	private static function update_default_data_for_ecommerce( $data ) {
+		// update default data with ecommerce data
+		$data['data']['topPriority']['priority1'] = 'selling';
+		$data['data']['siteType'] = array('label' => '', 'referTo' => 'business');
+		return $data;
+	}
+
+    /*
+	 * function to search for key in array recursively with case sensitive exact match
+	 */
+	public static function check_key_in_nested_array( $params, $flow_data, $header_key = 'Base Level' ) {
+		static $mismatch_key = [];
```

Comment:
```suggestion
		$mismatch_key = [];
```

Reply:
This is still not resolved @Yashita101

Reply:
This would remain static, else it is not able to detect the mismatch key in the nested arrays